packages/web-components/src/checkbox/checkbox.styles.ts

```diff
@@ -0,0 +1,182 @@
+    justify-content: center;
+    overflow: hidden;
+    margin: ${spacingVerticalS} ${spacingHorizontalS};
+    fill: currentcolor;
+    height: 16px;
+    width: 16px;
+    border: ${strokeWidthThin} solid ${colorNeutralStrokeAccessible};
+    border-radius: ${borderRadiusSmall};
+    outline: none;
+  }
+  .label {
+    align-self: center;
+    cursor: inherit;
+    font-family: ${fontFamilyBase};
+    font-size: ${fontSizeBase300};
+    line-height: ${lineHeightBase300};
+    color: ${colorNeutralForeground3};
+    padding: ${spacingVerticalS} ${spacingHorizontalS} ${spacingVerticalS} ${spacingHorizontalXS};
+  }
+  .label__hidden {
+    display: none;
+    visibility: hidden;
+  }
+  .checked-indicator {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    color: ${colorNeutralForegroundInverted};
+    height: 16px;
+    font-size: 12px;
+    margin: auto;
+    opacity: 0;
+  }
+  .indeterminate-indicator {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    width: 8px;
+    height: 8px;
+    transform: translate(-50%, -50%);
+    border-radius: ${borderRadiusSmall};
+    background-color: ${colorCompoundBrandForeground1};
+    opacity: 0;
+  }
+  :host(:hover) .indeterminate-indicator {
+    background-color: ${colorCompoundBrandForeground1Hover};
+  }
+  :host(:hover) .label {
+    color: ${colorNeutralForeground2};
+  }
+  :host(:hover) .control {
+    border-color: ${colorNeutralStrokeAccessibleHover};
+  }
+  :host(:active) .control {
+    border-color: ${colorNeutralStrokeAccessiblePressed};
+  }
+  :host(:active) .indeterminate-indicator {
+    background-color: ${colorCompoundBrandForeground1Pressed};
+  }
+
+  :host([aria-checked='true']) .label,
+  :host([aria-checked='mixed']) .label,
+  :host(:active) .label {
+    color: ${colorNeutralForeground1};
+  }
+  :host([aria-checked='true']) .control {
+    background-color: ${colorCompoundBrandBackground};
+  }
+  :host([aria-checked='true']:hover) .control {
+    border-color: ${colorCompoundBrandStrokeHover};
+    background-color: ${colorCompoundBrandBackgroundHover};
+  }
+  :host([aria-checked='true']:active) .control {
+    background-color: ${colorCompoundBrandBackgroundPressed};
+  }
+  :host([aria-checked='mixed']:hover) .control {
+    border-color: ${colorCompoundBrandStrokeHover};
+  }
+  :host([aria-checked='true']) .checked-indicator,
+  :host([aria-checked='mixed']) .indeterminate-indicator {
+    opacity: 1;
+  }
+  :host([aria-checked='true']) .control,
+  :host([aria-checked='mixed']) .control {
+    border-color: ${colorCompoundBrandStroke};
+  }
+  :host([aria-checked='mixed']:active) .control,
+  :host([aria-checked='true']:active) .control {
+    border-color: ${colorCompoundBrandStrokePressed};
+  }
+
+  :host([label-position='before']) {
+    flex-direction: row-reverse;
+  }
+  :host([label-position='before']) .label {
+    padding: ${spacingVerticalS} ${spacingHorizontalXS} ${spacingVerticalS} ${spacingHorizontalS};
```

Comment:
Your left and right paddings are different, you'll need to use `padding-inline` for `rtl`